General Changes

lejshi3 opened the pull request:
I replaced the black squares on the home page with semi-related placeholders using images from the CC tutorial screens. These changes are placeholders and not meant to be final, they're just supposed to make putting in new images in their place easier in the future, also they look nicer than black squares.

![image](https://user-images.githubusercontent.com/45546025/99544571-14675f80-297a-11eb-9098-b24997c342a4.png)

***Note: If this Pull Request is Accepted, Please Change the `"Homepage": "https://sirjoshi.github.io/Cortex-Command-Community.github.io/",` with `"Homepage": "Cortex-Command-Community.github.io"` or delete the line altogether or else it will show a blank webpage when deployed.***

lejshi3 committed:
beautified the index.html

facebook robot code bad,
indented human code good.

lejshi3 committed:
Replaced the Placeholders with Placeholders

I replaced the black squares with pictures from the game's tutorial files.

I would not call these changes final, but it will save some time for when someone has new pictures/gifs to go here.

the quest for the pink role continues

lejshi3 committed:
Merge branch 'master' of https://github.com/SirJoshi/Cortex-Command-Community.github.io

lejshi3 committed:
Update index.html

lejshi3 committed:
Update index.html

lejshi3 committed:
mmmmmm

i am in pain, i am going to make a pull request soon

lejshi3 committed:
jhmmmm

maybe this will work?

lejshi3 committed:
idk anymore

lejshi3 committed:
Update package.json

lejshi3 committed:
Update index.html

lejshi3 committed:
please let this let it build on gh

added alt text

lejshi3 committed:
Update package.json

lejshi3 committed:
Update package.json

lejshi3 committed:
Update package.json

garethyr commented on package.json:
```diff
@@ -2,6 +2,7 @@
   "name": "cortex",
   "version": "0.1.0",
   "private": true,
+  "homepage": "https://sirjoshi.github.io/Cortex-Command-Community.github.io/",
```
It's not a big deal, but you should do a little cleanup - in your dev environment, change this yourself and commit it to your relevant branch so it doesn't show up as cruft here.

garethyr commented on public/index.html:
```diff
@@ -28,7 +28,7 @@
   </head>
   <body>
-    <noscript>You need to enable JavaScript to run this app.</noscript>
+    <noscript>You need to enable JavaScript to view this page, nerd.</noscript>
```
Undo this please and thanks.

garethyr commented on src/components/Home/Home.js:
```diff
@@ -7,6 +7,12 @@ import CortexCardContainer from './CortexCardContainer/CortexCardContainer';
 import CortexCardContents from './CortexCardContents/CortexCardContents';
 import MagicArrow from './MagicArrow/MagicArrow';
 
+/*i pray to god (if there is one) that there is a beter way to go about this than just manually importing every images*/
```
Kill off this comment please.
For your future knowledge, a good way to do this sort of thing is leave your own comment on the PR, explaining that you're unsure of whether this is the right way to do something. It means cleaner code and it's pretty much impossible for reviewers to miss.

garethyr commented on src/components/Home/Home.js:
```diff
@@ -27,19 +33,24 @@ function Home() {
                             title='Upgraded Engine'
                             body='The source code has been enhanced to improve performance.'
+                            image={DummySquadImg}
+                            alt='some dummies'
```
Please change the alt text slightly so it's a little better/more uniform:
`some dummies` -> `Dummy Squad Image`
`a buy menu` -> `Buy Menu Image`
`a joystick` -> `Joystick Image`

garethyr reviewed (changes requested):
I'll leave code structure comments to Tom, here's some cleanup for you to do while you wait for his review.

lejshi3 committed:
Update Home.js

made much more descriptive alt texts

lejshi3 committed:
Update index.html

professionalism

lejshi3 committed:
Update package.json

deleted the homepage property

lejshi3 commented:
Some changes have been made in the name of Accessibility and Professionalism

lejshi3 committed:
Update Home.js

lejshi3 committed:
indentation

not important, just was bugging me

lejshi3 committed:
Lots of Changes

1. Modified the Page to be Dark not Light (Light is gross)

2. Made the 404 Page Cooler

3. Made the Downloads have a direct link to Pre2 and a Get Mods section that links to the Discord and the Mod.io

4. Modified the Downloads page to make the border go even further arround the text (it just looks cooler)

lejshi3 reviewed (commented):
### Changes:

1. I changed a lot of CSS and Style to make the Pages Dark as Opposed to Light

2. I added placeholder images to the home page to make adding in new pics/gifs easier

3. I no longer call js-phobics nerds

4. I made some subtle changes to the style of the Downloads page

5. I added a direct link to Pre 2 in the Downloads page

6. I added a Get Mods section to the Downloads page that links to the Discord and the Mod.io

tomflenniken commented on src/components/Home/CortexCardContents/CortexCardContents.js:
```diff
@@ -13,15 +13,18 @@ function CortexCardContents(props) {
             </div>
             {/* This is an image placeholder **Imagine an Image** */}
-            <div style={{
+        <img style={{
```
I believe the original indentation was correct here.

tomflenniken commented on src/components/Home/Home.js:
```diff
@@ -27,19 +32,25 @@ function Home() {
                             title='Improved Strategic Gameplay'
                             body='The team has focused on accentuating the most satisfying gameplay elements.'
+                            image={JoystickImg}
+                            alt='A Handdrawn/Pixelart Image of a Joystick, an input method that can be used to (ineffiecently) play Cortex Command.'
```
probably too long for alt text - though I appreciate the levity.
Maybe cut it to `A pixel art image of a joystick.`

Also I think sentence case is appropriate for these alt texts.
I would also avoid using "/" because a primary use for alt text is screen readers, and I don't think they'd read that out in a natural sounding way.

Thanks for providing alt text, it's probably not super critical for a site for a game that has 0 accessibility features, but it helps for making the web a bit more inclusive.

tomflenniken commented on src/components/Static/NotFound/NotFound.js:
```diff
@@ -1,12 +1,20 @@
             <h1 style={{ marginTop: '48px' }}>Page Not Found</h1>
             <p>Sorry, but the page you were trying to view does not exist.</p>
+            <br />
+            <br />
```
I'm not wholly opposed to using `<br />` here (I might have used it some places myself), but generally I think using CSS for spacing should be preferred.

Don't change it if you don't feel like it.

tomflenniken commented on src/components/Static/NotFound/NotFound.js:
```diff
@@ -1,12 +1,20 @@
             <p>Sorry, but the page you were trying to view does not exist.</p>
+            <br />
+            <br />
+            <a href='https://www.youtube.com/watch?v=t0s9dRNjhfg?autoplay=1'><img style={{postion: 'absolute', width: '25%'}} src={EmojiDunno}/></a>
```
Interesting easter egg.  Doesn't seem super _relevant_ in this context, but I'm fine with it.

tomflenniken commented on src/components/App/App.css:
```diff
@@ -1,6 +1,6 @@
 body {
-  color: #111;
-  background-color: #eee
+  color: #EEE;
```
I like the light theme :(

tomflenniken replied:
Actually, can you pull out the theme edits into a separate branch? I think it needs to be addressed separately. I think it's a good opportunity to adjust the stylesheets and how style is managed across the site.
See #9

tomflenniken commented on src/components/Home/Home.js:
```diff
@@ -7,6 +7,11 @@ import CortexCardContainer from './CortexCardContainer/CortexCardContainer';
+/*image imports*/
+import DummySquadImg from './img/Team.png'
+import JoystickImg from './img/Joystick.png'
+import BuymenuImg from './img/BuyMenu.png'
```
probably `BuyMenuImg`

These can alternatively be included in `/public` in which case they won't be base64 encoded. I think I'm doing it both ways currently (probably a bad idea) for the existing images.

tomflenniken commented on src/components/Static/Downloads/Downloads.js:
```diff
@@ -3,27 +3,45 @@ import Template from '../../Template/Template';
         <p style={{
-          marginBottom: '18px',
-        }}>This is where you can find the community's plans for Cortex Command's future. There'll be rebalancing, new weapons and units, and much more.</p>
+          marginTop: '8px', marginBottom: '18px',
```
I don't know if there's any current consistency across the site on a 6px vs 8px based unit size, but it's probably worth being consistent at least internally. Eventually we'd want to move to being able to do something like,
```
marginTop: unit(1),
marginBottom: unit(3)
```
or whatever.

tomflenniken replied:
So the takeaway here is to choose which looks better:
`marginTop: '8px', marginBottom: '16px',`
or
`marginTop: '6px', marginBottom: '18px',`

garethyr replied:
Definitely agree that setting up standardized sizes is the way to go. Probably too late to be convenient now, but using rem sizes as opposed to px sizes is quite nice.

tomflenniken commented on src/components/Static/Downloads/Downloads.js:
```diff
@@ -3,27 +3,45 @@ import Template from '../../Template/Template';
+      
         <Button link to='https://github.com/cortex-command-community/Cortex-Command-Community-Project-Data/releases'>Releases</Button>
+      
+        <Button link to='https://github.com/cortex-command-community/Cortex-Command-Community-Project-Data/releases/download/v0.1.0-pre2/CCCP.v0.1.0-pre2.zip'>Pre 2 <img width='14px' height='14px' marginBottom='-12px' src={EmojiGood}/> (latest)</Button>
```
I like including this link here. Not sure about the icon though. What's your thinking on that?

garethyr replied:
I'm also on the fence about the icon - it's a cute idea but I think it looks a little odd.

tomflenniken commented on src/components/Static/Downloads/Downloads.js:
```diff
@@ -3,27 +3,45 @@ import Template from '../../Template/Template';
+      
+      </Card>
+      <h2>Mods</h2>
+      <Card title={'Get Mods'}>
```
Definitely an important addition. 
Might be useful to know what the plan is to allow users to find the correct mod versions?

tomflenniken commented on src/components/Static/NotFound/NotFound.js:
```diff
@@ -1,12 +1,20 @@
 function NotFound() {
     return (
         <Template>
+            <div style={{textAlign: 'center'}}>
```
I think left aligned is more professional looking, at least for text.

tomflenniken commented on src/components/Template/Template.js:
```diff
@@ -12,8 +12,10 @@ function Template(props) {
       <Header />
       <div style={{
         margin: 'auto',
+        marginBottom: '0px',
```
What are the changes in this file for?

tomflenniken commented on src/components/Home/Home.js:
```diff
@@ -7,6 +7,11 @@ import CortexCardContainer from './CortexCardContainer/CortexCardContainer';
 import CortexCardContents from './CortexCardContents/CortexCardContents';
 import MagicArrow from './MagicArrow/MagicArrow';
 
+/*image imports*/
```
I think prettier or something enforces import ordering, so I typically just put all the imports together so the autoformatter can do its thing.

tomflenniken commented on src/components/Static/Downloads/Downloads.js:
```diff
@@ -3,27 +3,45 @@ import Template from '../../Template/Template';
+        <Button link to='https://github.com/cortex-command-community/Cortex-Command-Community-Project-Data/releases/download/v0.1.0-pre2/CCCP.v0.1.0-pre2.zip'>Pre 2 <img width='14px' height='14px' marginBottom='-12px' src={EmojiGood}/> (latest)</Button>
+      
+      </Card>
+      <h2>Mods</h2>
```
With the addition of some more content on this page I think it's looking a little cluttered. No action required now, but I will probably be restructuring some of the headings on this page/grouping up some of the card content.

tomflenniken commented on src/components/Static/Downloads/Downloads.js:
```diff
@@ -3,27 +3,45 @@ import Template from '../../Template/Template';
+      
         <Button link to='https://github.com/cortex-command-community/Cortex-Command-Community-Project-Data/releases'>Releases</Button>
+      
+        <Button link to='https://github.com/cortex-command-community/Cortex-Command-Community-Project-Data/releases/download/v0.1.0-pre2/CCCP.v0.1.0-pre2.zip'>Pre 2 <img width='14px' height='14px' marginBottom='-12px' src={EmojiGood}/> (latest)</Button>
```
`marginBottom` here is giving a console error. It should be defined as part of a style attribute
```style={{ marginBottom= ... }}```

tomflenniken reviewed (commented):
Looking good, I definitely like the new placeholder images. I agree gifs would be nice, but this is a great start.

tomflenniken commented:
![responsive buttons](https://user-images.githubusercontent.com/20175921/99607227-3683d080-29c0-11eb-9ff3-5522a8c45ba4.png)
I noticed a responsiveness issue for the buttons that should be addressed.

lejshi3 committed:
Misc Changes

A New Picture and a New Button

lejshi3 committed:
get involved page

lejshi3 committed:
made it more cortexy

lejshi3 committed:
Merge https://github.com/cortex-command-community/Cortex-Command-Community.github.io

lejshi3 committed:
download page

lejshi3 committed:
stuff and things

lejshi3 committed:
changed the images into gifs

lejshi3 commented:
Okay, Hello, Happy 2021.

A lot of the issues above have been addressed so i'll keep this brief.
- Added GIFS
- Added Get Involved
- Expanded on the Downloads Page

After the PR, I'd consider the website in a usable state.

tomflenniken commented:
@SirJoshi can you split the changes out into separate branches for the different pages that they impact (or any other reasonable way to break down the changes here)?

I think we should tackle these in a more targeted manner, I think it will be easier to get your changes through in chunks.

lejshi3 committed:
Update package.json
